src/representations/calculator_spherical_invariants.hh

```diff
@@ -1277,29 +1277,6 @@ namespace rascal {
         soap_vector_gradients[ii_pair].resize(
             pair_list, n_spatial_dimensions * n_row, n_col, 0.);
 
-        // TODO(max,felix) needs work
-        /*
-        // Neighbour gradients need a separate pair list because if the species
-        // of j is not the same as either of the species for that SOAP entry,
-        // the gradient is zero.
-        for (auto neigh : center) {
-          std::vector<internal::SortedKey<Key_t>> grad_pair_list{};
-          for (const auto & el1 : coefficients) {
-            auto && neigh_1_type{el1.first[0]};
-            for (const auto & el2 : coefficients) {
-              auto && neigh_2_type{el2.first[0]};
-              auto neigh_type = neigh.get_atom_type();
-              if (neigh_1_type <= neigh_2_type) {
-                pair_type[0] = neigh_1_type;
-                pair_type[1] = neigh_2_type;
-                if ((neigh_type == pair_type[0]) or
-                    (neigh_type == pair_type[1])) {
-                  grad_pair_list.emplace_back(is_sorted, pair_type);
-                }
-              }
-            }
-          }
-          */
```

Comment:
It is already open and we keep the code here to be able to restart easily.

Reply:
Is this #115?

Reply:
I moved the code there then

Reply:
could you also undo the delete from this file ?

Reply:
There is no need to keep it there anymore, right?

Reply:
it is at the right place to start working on the issue again. I don't remember that leaving this breaks the code convention so please put it back there.

Reply:
I am sorry but I disagree. Keeping commented out code in git just adds clutter. How about an external point of view on this? Ping @tjunge @mastricker @ceriottm =)

Reply:
I agree with @Luthaf , the more we move into the direction of a release, the code base should be clean. And any time is as good as now to start working towards a cleaner codebase. We are now ~8 people working on the same code base, not 2-3 like half a year/a year ago. If everyone now keeps reminders inside there...well you see where this is going.

And there is a mechanism to keep this information: branch it into a reminder branch and everything is there, once it is needed, not? I am totally fine with a comment at that section referring to a branch name where the needed information can be recovered - is that a viable solution?

Something like
`// restart work with help of branch feat/reminder_soap_stuff`
and make an issue, so we see it from time to time.

I actually like the continuous monitoring of the issues and branches we are having now. I feel it is a small price to pay to keep Issues and branches around to keep track of things and routinely going over them to clarify. Some things settle themselves or become irrelevant over time.

Reply:
then you should make this requirement part of the code convention in the next meeting. And again I don't see how it clutters the code. I guess you are much more tidy than I am.

Reply:
aaargh ffs I tried to find the commit that changed this (so I could see where exactly it was being deleted) but the bleeding rebase got rid of it, so the web interface can no longer find it. Please don't be so aggressive with the rebases and the force-pushes; it doesn't matter that it's on your own repo, we're still collaborating and commenting on it, so force-pushes disrupt the entire conversation about the proposed changes.

Reply:
As for deleting this code block: I'm also against it; it's a proposed implementation of a more efficient gradient computation that we just haven't had the chance to work on yet (though it will need to be done in time for the release). At the _very_ least you need to keep the comment that says this isn't the most efficient way of doing things, and that this is the place to make those changes.

For proposed implementations like this, the less we have to swap in to remember our previous starting point, the better. I don't need to spend 5 minutes looking for the issue that had this code, copying and pasting it back in, when I could just start working on it right away.

Reply:
I second you on all your points @max-veit

Reply:
> For proposed implementations like this, the less we have to swap in to remember our previous starting point, the better. I don't need to spend 5 minutes looking for the issue that had this code, copying and pasting it back in, when I could just start working on it right away.

I agree with you here, my point is more that such work-in-progress code should not live as commented out code in master, but as compilable code inside a separated branch.

I'll put this one back so that we can move forward with the PR, but I would prefer having a clean master branch as much as possible.